kratos/tests/test_matrix_interface.py

```diff
@@ -120,5 +120,39 @@ def test_matrix_transpose(self):
             for j in range(A.Size2()):
                 self.assertEqual(A[i,j], B[j,i])
 
+    def test_empty_list_construction(self):
+        A = KM.Matrix([])
+        self.assertMatrixAlmostEqual(A, KM.Matrix(0,0))
+
+    def test_empty_list_of_list_construction(self):
+        B = KM.Matrix([[]])
+        self.assertMatrixAlmostEqual(B, KM.Matrix(0,0))
+
+    def test_rectangular_list_of_list_construction(self):
+        C = KM.Matrix([[2, 2, 2, 2]])
+        self.assertEqual(4, len(C))
+        self.assertEqual(1, C.Size1())
+        self.assertEqual(4, C.Size2())
+
+        for i in range(C.Size1()):
+            for j in range(C.Size2()):
+                self.assertEqual(C[i,j], 2)
+
+    def test_sqaure_list_of_list_construction(self):
+        l2 = [[0, 0.1, 0.2], [1, 1.1, 1.2], [2, 2.1, 2.2]]
+        D = KM.Matrix(l2)
+        self.assertEqual(9, len(D))
+        self.assertEqual(3, D.Size1())
+        self.assertEqual(3, D.Size2())
+
+        for i in range(D.Size1()):
+            for j in range(D.Size2()):
+                self.assertEqual(D[i,j], i + (j*0.1))
+
+    def test_list_of_list_construction_error_hand(self):
+        with self.assertRaisesRegex(RuntimeError, r'Error: Wrong size of a row'):

```

Comment:
@philbucher Is this a correct way to test an exception ???

Reply:
yes!
Needs minor change after my suggestion